modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -586,8 +590,174 @@ private void doColumnAutoSize(TableColumnBase<?,?> column, int cellsToMeasure) {
 
         // if the prefWidth has been set, we do _not_ autosize columns
         if (prefWidth == DEFAULT_COLUMN_WIDTH) {
-            TableSkinUtils.resizeColumnToFitContent(getTableSkin(), column, cellsToMeasure);
-//            getTableViewSkin().resizeColumnToFitContent(column, cellsToMeasure);
+            resizeColumnToFitContent(column, cellsToMeasure);
+        }
+    }
+
+    /**
+     * Resizes the given column based on the preferred width of all items contained in it. This can be potentially very
+     * expensive if the number of rows is large. Subclass can either call this method or override it (no need to call
+     * {@code super()}) to provide their custom algorithm.
+     *
+     * @param tc      the column to resize
+     * @param maxRows the number of rows considered when resizing. If -1 is given, all rows are considered.
+     * @since 12
+     */
+    protected void resizeColumnToFitContent(TableColumnBase<?, ?> tc, int maxRows) {
+        if (!tc.isResizable()) return;
```

Comment:
see https://github.com/javafxports/openjdk-jfx/pull/289#discussion_r245482258 - this api still looks unhealthy ..